### ISSUE TITLE
escaping asterisk not working

### ISSUE THREAD

**Issue by leewjae** [projectnessie/cel-java#284]
Hello!

I am using cel-java and when I try to escape \* it gives me an error.
for example i want to get entry by using filter=entry.name.matches('a\\*'), but nessie responses with :
```
{
	"status": 400,
	"reason": "Bad Request",
	"message": "org.projectnessie.cel.tools.ScriptCreateException: parse failed: ERROR: <input>:1:20: Syntax error: token recognition error at: '\"a\\*'\n | entry.name.matches(\"a\\*\")\n | ...................^\nERROR: <input>:1:24: Syntax error: token recognition error at: '\")'\n | entry.name.matches(\"a\\*\")\n | .......................^\nERROR: <input>:1:26: Syntax error: mismatched input '<EOF>' expecting {'[', '{', '(', ')', '.', '-', '!', 'true', 'false', 'null', NUM_FLOAT, NUM_INT, NUM_UINT, STRING, BYTES, IDENTIFIER}\n | entry.name.matches(\"a\\*\")\n | .........................^",
	"errorCode": "BAD_REQUEST",
	"serverStackTrace": null
}
```
Thanks!

**Comment by XN137** [comment]
replacing the `\n` with actual newlines in the error:
```
org.projectnessie.cel.tools.ScriptCreateException: parse failed: ERROR: <input>:1:20: Syntax error: token recognition error at: '\"a\\*'
 | entry.name.matches(\"a\\*\")
 | ...................^
ERROR: <input>:1:24: Syntax error: token recognition error at: '\")'
 | entry.name.matches(\"a\\*\")
 | .......................^
ERROR: <input>:1:26: Syntax error: mismatched input '<EOF>' expecting {'[', '{', '(', ')', '.', '-', '!', 'true', 'false', 'null', NUM_FLOAT, NUM_INT, NUM_UINT, STRING, BYTES, IDENTIFIER}
 | entry.name.matches(\"a\\*\")
 | .........................^
```

i am wondering:
how did you arrive at the conclusion that you need to escape the `*` char in the first place?
i dont think this is necessary at all.

to test things in a nessie context we can modify the test here:
https://github.com/projectnessie/nessie/blob/d7d7f5ef6b35f5c36b663a529f0365c3ca469a54/servers/jax-rs-tests/src/main/java/org/projectnessie/jaxrs/AbstractRestEntries.java#L121-L128

like this:
```
    entries =
      getApi()
        .getEntries()
        .reference(refMode.transform(branch))
        // .filter("entry.name.matches('first.\\*')") // FAILS due to invalid escape sequence
        // .filter("entry.name.matches(\"first.\\*\")") // FAILS due to invalid escape sequence
        // .filter("entry.name.matches('first.*')") // WORKS
        .filter("entry.name.matches(\"first.*\")") // WORKS
        .get()
        .getEntries();
    assertThat(entries.stream().map(Entry::getName)).containsExactly(first);
```
and the last 2 filter strings work just fine.
(note that using `'` makes things easier when writing string literals in cel from java code or json, since there `"` is already used and needs to be escaped)

string literals are explained in the `cel-spec` here:
https://github.com/google/cel-spec/blob/master/doc/langdef.md#string-and-bytes-values
and the escaping mechanism is explained there as well,
**so basically `\*` is an invalid escape sequence causing the parsing error**.

if you wanted to, you could probably express `*` with its unicode codepoints as well but as mentioned this is not necessary.

side note 1:
the regular expression `a*` matches any string that contains `a` zero or more times (so basically any string)
i dont think this is what you want, see:
https://github.com/google/cel-spec/blob/master/doc/langdef.md#regular-expressions
https://github.com/google/re2/wiki/Syntax

so if you want to match a string starting with `a` then it would be `^a.*`

side note 2:
there is also `entry.name.startsWith('a')` for example if that is what you want to do

**Comment by XN137** [comment]
i just realized, maybe you want to find strings that contain the literal `a*` ?
then you should be able to use `entry.name.contains('a*')`

**Comment by leewjae** [comment]
Issue was due to CEL string literals, using raw strings resolves the issue
